tests/testTriggers.go

```diff
@@ -7195,6 +7197,159 @@ func TriggerKafkaAsyncDR(contexts *[]*scheduler.Context, recordChan *chan *Event
+		if err != nil {
+			UpdateOutcome(event, fmt.Errorf("Failed to write kubeconfig, err: %v", err))
+			return
+		}
+		err = SetSourceKubeConfig()
+		if err != nil {
+			UpdateOutcome(event, fmt.Errorf("Failed to set source config, err: %v", err))
+			return
+		}
+	})
+	stepLog = fmt.Sprintf("Deploy applications with %v chaos level", chaosLevel)
+	Step(stepLog, func() {
+		log.InfoD(stepLog)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			//taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			ns, err := core.Instance().GetNamespace(nsName)
+			if err != nil {
+				log.InfoD("Creating namespace %v", nsName)
+				nsSpec := &v1.Namespace{
+					ObjectMeta: meta_v1.ObjectMeta{
+						Name: nsName,
+					},
+				}
+				ns, err = core.Instance().CreateNamespace(nsSpec)
+				if err != nil {
+					UpdateOutcome(event, fmt.Errorf("Failed to create namespace, err: %v", err))
+					return
+				}
+
+			}
+			log.InfoD("Pods Creation Started")
+			pods_created, err := asyncdr.HelmRepoAddandCrInstall(repoName, "https://mongodb.github.io/helm-charts", ns.Name, operatorName, fmt.Sprintf("%v/%v", repoName, operatorName), appUrl)
+			if err != nil {
+				UpdateOutcome(event, fmt.Errorf("Failed to create app pods, err: %v", err))
+				return
+			}
+			pods_created_len := len(pods_created.Items)
+			log.InfoD("Num of Pods on source: %v", pods_created_len)
+			sourceClusterConfigPath, err := GetSourceClusterConfigPath()
+			if err != nil {
+				UpdateOutcome(event, fmt.Errorf("Failed to get cluster config path: %v", err))
+				return
+			}
+			err = asyncdr.ValidateCRD(asyncdr.ExpectedMongoCrdList, sourceClusterConfigPath)
+			if err != nil {
+				UpdateOutcome(event, fmt.Errorf("CRD validation failed on source, err: %v", err))
+				return
+			}
+			options := scheduler.ScheduleOptions{Namespace: ns.Name}
+			var emptyCtx = &scheduler.Context{
+				UID:             "",
+				ScheduleOptions: options,
+				App: &spec.AppSpec{
+					Key:      "",
+					SpecList: []interface{}{},
+				}}
+			stepLog = "Create cluster pair between source and destination clusters"
+			Step(stepLog, func() {
+				log.InfoD(stepLog)
+				// Set cluster context to cluster where torpedo is running
+				log.InfoD("ClusterPairing Started")
+				ScheduleValidateClusterPair(emptyCtx, false, true, defaultClusterPairDir, false)
+			})
+			stepLog = "Start migration and validate"
+			Step(stepLog, func() {
+				log.InfoD(stepLog)
+				log.InfoD("Migration Started")
+				mig_name := migrationKey + "mongo-" + fmt.Sprintf("%d", i) + time.Now().Format("15h03m05s")
+				mig, err := asyncdr.CreateMigration(mig_name, ns.Name, asyncdr.DefaultClusterPairName, ns.Name, &includeVolumesFlag, &includeResourcesFlag, &startApplicationsFlag)
+				if err != nil {
+					UpdateOutcome(event, fmt.Errorf("Failed to create migration: %v", err))
+					return
+				}
+				migrationList = append(migrationList, mig)
+				err = asyncdr.WaitForMigration(migrationList)
+				if err == nil {
```

Comment:
instead of err == nil check do err !=nil check and return if err
so that we can avoid using else

Reply:
Done